### ISSUE TITLE
Going into debt can make the game stop working

### ISSUE THREAD

**Issue by Lock-The-Door** [Lock-The-Door/Flowery-Fields#15]
```
You are in debt

Update shop visual: Shoes

InvalidOperationException: Sequence contains no matching element
  at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00011] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at Shop.UpdateBuyButtonVisual (ShopItem shopItem) [0x0002d] in <8187a2153d734036a96ed5850eeab04b>:0 
  at GameFlow+<FinishDay>d__30.MoveNext () [0x00212] in <8187a2153d734036a96ed5850eeab04b>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) [0x00026] in <20392e72525040018eca4991d4519e65>:0
```

**Comment by Lock-The-Door** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Going into debt can make the game stop working](https://trello.com/c/Iso0OVgR/54-going-into-debt-can-make-the-game-stop-working)

**Comment by Lock-The-Door** [comment]
Recreation steps: 
1. Exit game while in debt with something bought
2. Finish day until the item needs to be sold
3. The game gets stuck

**Comment by Lock-The-Door** [comment]
Solved in commit aa260595bf13656d2ec57a3cb3ede1f1fcaad69a will be merged in #19
